src/screens/GroupSetting/hooks/useGroupSetting.js

```diff
@@ -36,53 +54,71 @@ const useGroupSetting = ({navigation, route}) => {
     }
   }, [route]);
 
-  const submitData = async (withNavigation = true, withLoading = true) => {
-    let changeImageUrl = '';
-    if (changeImage) {
-      if (withLoading) setIsLoading(true);
-      try {
-        const res = await uploadFile(`data:image/jpeg;base64,${base64Profile}`);
-        changeImageUrl = res.data.url;
-      } catch (e) {
-        setIsLoading(false);
-        return;
-      }
+  React.useEffect(() => {
+    if (selectedChannel) {
+      setUrlImage(selectedChannel?.channelPicture);
     }
+  }, [selectedChannel]);
 
-    if (changeName || changeImage) {
-      if (withLoading) setIsLoading(true);
-      let dataEdit = {
-        name: groupName
-        // ...(changeImage && {image: base64Profile}),
+  const submitData = async (withNavigation = true) => {
+    try {
+      setLoadingUpdate(true);
+      let body = {
+        channel_id: route?.params?.id,
+        channel_name: groupName
       };
-      if (changeName) {
-        dataEdit = {...dataEdit, isEditName: true};
+      if (urlImage) {
+        try {
+          const res = await uploadFile(`data:image/jpeg;base64,${base64Profile}`);
+          body = {...body, channel_image: res.data.url};
+        } catch (e) {
+          console.log(e, 'error upload');
+        }
       }
-      if (changeImage) {
-        dataEdit.image = changeImageUrl;
-      } else if (channel?.data?.image) {
-        dataEdit.image = channel?.data?.image;
+      const response = await SignedMessageRepo.editChannel(body);
+      let newChannel = response.data?.channel;
+      newChannel = {
+        ...chat.selectedChannel,
+        ...newChannel,
+        channelType: handleChannelType(newChannel?.channelType),
+        channelPicture: body?.channel_image
+      };
+      ChannelList.updateChannelInfo(
+        localDb,
+        body.channel_id,
+        body.channel_name,
+        body.channel_image
+      );
+      if (newChannel) {
+        setChat({
+          ...chat,
+          selectedChannel: newChannel
+        });
       }
-      console.log({dataEdit}, 'nusu');
-      updateDataEdit(dataEdit, withNavigation);
-    } else if (withNavigation) navigation.goBack();
-  };
+      refresh('channelInfo');
+      refresh('chat');
 
-  const updateDataEdit = async (dataEdit, withNavigation) => {
-    try {
-      await channel.update(dataEdit);
-      if (withNavigation) navigation.navigate('ChatDetailPage');
-    } catch (e) {
-      if (__DEV__) {
-        console.log(`error : ${e}`);
+      refresh('channelList');
+
+      if (withNavigation) {
+        navigation.goBack();
       }
+      setLoadingUpdate(false);
+    } catch (e) {
+      setLoadingUpdate(false);
       SimpleToast.show(StringConstant.groupSettingUpdateFailed);
     }
-    setIsLoading(false);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `submitData` function, there's an error handling issue. When an error occurs during file upload, the function logs the error but continues execution. This could lead to unexpected behavior if the file upload is critical to the rest of the function. Consider throwing the error or returning from the function after logging the error.

```diff
      if (urlImage) {
        try {
          const res = await uploadFile(`data:image/jpeg;base64,${base64Profile}`);
          body = {...body, channel_image: res.data.url};
        } catch (e) {
          console.log(e, 'error upload');
+          throw e;
        }
       }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->